### PR TITLE
Add AuthorizeWith support off ConnectionBuilder

### PR DESCRIPTION
**Pull request by benmccallum**
Provides ability to do the following:
```c#
Connection<MyType>()
    .Name("myTypes")
    .AuthorizeWith("SomePolicy")
    .Resolve(...);
```

### CODE REVIEW

**Comment by benmccallum** [comment]
cc: @RehanSaeed

**Comment by benmccallum** [comment]
@sungam3r, I've directed this at master, since it can go to master and develop easily enough.

**Comment by benmccallum** [comment]
(assuming `develop` supports the same ConnectionBuilder, it may have changed in 3.0

**Comment by RehanSaeed** [comment]
LGTM

**Comment by sungam3r** [comment]
I wanted to review this on the weekend, but I did not have time. Will definitely do it later.

**Comment by benmccallum** [comment]
No worries :)

**Comment by benmccallum** [comment]
Looks like for graphql-dotnet v3.0 this needs to be just the following:
```
        public static ConnectionBuilder<TSourceType> AuthorizeWith<TSourceType>(
            this ConnectionBuilder<TSourceType> builder, string policy)
        {
            builder.FieldType.AuthorizeWith(policy);
            return builder;
        }
```

**Comment by benmccallum** [comment]
*reminder to look at this later*

**Comment by benmccallum** [comment]
This is now ready. I've updated it to point to develop and be compatible there.

File: tests/Authorization.AspNetCore.Tests/AuthorizationValidationRuleTests.cs [hunk 1]
```diff
@@ -152,6 +153,43 @@ public void fails_on_missing_claim_on_input_type()
...
+                    options.AddPolicy("ConnectionPolicy", x => x.RequireClaim("admin"));
+                });
+
+            ShouldPassRule(_ =>
+            {
+                _.Query = @"query { posts { items { id } } }";
+                _.Schema = TypedSchema();
+                _.User = CreatePrincipal(claims: new Dictionary<string, string>
+                {
+                    {"Admin", "true"}
```
Comment (sungam3r) [inline tests/Authorization.AspNetCore.Tests/AuthorizationValidationRuleTests.cs]:
```suggestion
                    { "Admin", "true" }
```

**Comment by sungam3r** [comment]
@benmccallum I finally found the time to do a review

**Pull request merged**